http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcProviderMetadataTest.java

```diff
@@ -0,0 +1,368 @@
+            ",\"request_parameter_supported\":true\n" +
+            ",\"request_uri_parameter_supported\":true\n" +
+            ",\"pushed_authorization_request_endpoint\":\"http://localhost:8080/realms/myrealm/protocol/openid-connect/ext/par/request\"\n" +
+            ",\"revocation_endpoint\":\"http://localhost:8080/realms/myrealm/protocol/openid-connect/revoke\"\n" +
+            ",\"revocation_endpoint_auth_methods_supported\":[\"private_key_jwt\",\"client_secret_basic\",\"client_secret_post\",\"tls_client_auth\",\"client_secret_jwt\"]\n" +
+            ",\"revocation_endpoint_auth_signing_alg_values_supported\":[\"PS384\",\"RS384\",\"EdDSA\",\"ES384\",\"HS256\",\"HS512\",\"ES256\",\"RS256\",\"HS384\",\"ES512\",\"PS256\",\"PS512\",\"RS512\"]\n" +
+            ",\"backchannel_logout_supported\":true\n" +
+            ",\"backchannel_logout_session_supported\":true\n" +
+            ",\"code_challenge_methods_supported\":[\"plain\",\"S256\"]\n" +
+            ",\"tls_client_certificate_bound_access_tokens\":true\n" +
+            ",\"request_object_encryption_enc_values_supported\":[\"A256GCM\",\"A192GCM\",\"A128GCM\",\"A128CBC-HS256\",\"A192CBC-HS384\",\"A256CBC-HS512\"]\n" +
+            ",\"request_object_encryption_alg_values_supported\":[\"RSA-OAEP\",\"RSA-OAEP-256\",\"RSA1_5\"]\n" +
+            "}";
+
+    private static final String emptyProviderMetaData = "{}";
+
+    private static final String withoutBooleansProviderMetaData = "{\n" +
+            "\"issuer\":\"http://localhost:8080/realms/myrealm\"\n" +
+            ",\"authorization_endpoint\":\"http://localhost:8080/auth\"\n" +
+            ",\"token_endpoint\":\"http://localhost:8080/token\"\n" +
+            ",\"introspection_endpoint\":\"http://localhost:8080/introspect\"\n" +
+            ",\"userinfo_endpoint\":\"http://localhost:8080/openid-connect/userinfo\"\n" +
+            ",\"end_session_endpoint\":\"http://localhost:8080/openid-connect/logout\"\n" +
+            ",\"jwks_uri\":\"http://localhost:8080/openid-connect/certs\"\n" +
+            ",\"check_session_iframe\":\"http://localhost:8080/login-status-iframe.html\"\n" +
+            ",\"grant_types_supported\":[\"authorization_code\",\"implicit\"]\n" +
+            ",\"response_types_supported\":[\"code\",\"none\",\"id_token\",\"token\"]\n" +
+            ",\"subject_types_supported\":[\"public\",\"pairwise\"]\n" +
+            ",\"id_token_signing_alg_values_supported\":[\"PS384\"]\n" +
+            ",\"id_token_encryption_alg_values_supported\":[\"RSA-OAEP\",\"RSA-OAEP-256\",\"RSA1_5\"]\n" +
+            ",\"id_token_encryption_enc_values_supported\":[\"A256GCM\"]\n" +
+            ",\"userinfo_signing_alg_values_supported\":[\"PS384\",\"none\"]\n" +
+            ",\"request_object_signing_alg_values_supported\":[\"PS384\",\"none\"]\n" +
```

Comment:
Let's also remove the other options that were introduced in ELY-2584 since it's also optional for the provider to include these in the metadata:

* request_object_signing_alg_values_supported
* request_object_encryption_enc_values_supported
* request_object_encryption_alg_values_supported
* pushed_authorization_request_endpoint

Reply:
done